java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/KafkaConsumerProvider.java

```diff
@@ -0,0 +1,38 @@
+package io.lenses.java.streamreactor.connect.azure.eventhubs.source;
+
+import java.util.Properties;
+import java.util.concurrent.BlockingQueue;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+
+/**
+ * Provider for BlockingQueuedKafkaConsumers.
+ */
+@Slf4j
+public class KafkaConsumerProvider {
+  //Consumer need a unique client ID per thread
+  private static int id = 0;
+
+  /**
+   * Instantiates BlockingQueuedKafkaConsumer from given properties.
+   *
+   * @param consumerProperties Properties for Kafka Consumer
+   * @param recordBlockingQueue BlockingQueue for ConsumerRecords
+   * @return BlockingQueuedKafkaConsumer instance.
+   */
+  public BlockingQueuedKafkaConsumer createConsumer(Properties consumerProperties,
+      BlockingQueue<ConsumerRecords<String, String>> recordBlockingQueue) {
+    String clientId;
+    synchronized (EventHubsKafkaConsumerController.class) {
+      clientId = "KafkaEventHubConsumer#" + id++;
```

Comment:
let's consider two connector tasks running in parallel. 
Each one will run this code and for each the `id` starts at zero and thus create the same client id. The static state is not shared between tasks. If you want to control the id, you can control from the Connector:start by enriching the config to add the order/sequence.

Reply:
Yeah, if the classloaders are different then it has to be done in the other way. I amended the code